### PR TITLE
Initial step version

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [X] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [X] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### Context

Initial version of the dedicated Carthage cache step.

### Changes

Call the shared cache implementation in `go-steputils` using Carthage-specific cache key and cache paths

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
+	logger := log.NewLogger()
+	envRepo := env.NewRepository()
+	inputParser := stepconf.NewInputParser(envRepo)
+	cmdFactory := command.NewFactory(envRepo)
+	cacheStep := step.New(logger, inputParser, envRepo, cmdFactory)
```
Comment (vshah23) [inline main.go:24]: Move step creation to its own function (e.g., https://github.com/bitrise-steplib/steps-fastlane/blob/master/main.go#L56)

File: main.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-steplib/steps-restore-carthage-cache/step"
+)
+
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
```
Comment (vshah23) [inline main.go]: nit: Use `ExitCode` 😄 
https://github.com/bitrise-steplib/steps-fastlane/blob/master/main.go#L22

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+go 1.17
+
+require (
+	github.com/bitrise-io/go-steputils/v2 v2.0.0-alpha.17
+	github.com/bitrise-io/go-utils/v2 v2.0.0-alpha.13
+)
+
+require (
+	github.com/bitrise-io/go-utils v1.0.1 // indirect
```
Comment (vshah23) [inline go.mod:11]: I wonder where the indirect reference to v1 is coming from 🤔

**Review by vshah23 — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (ofalvai) [reply main.go:24]: Does it make sense in such a simple step though? The entire `main.go` file is 30 lines long, so I wouldn't refactor this now, but if the step gets more complex I'm happy to do it. Is that okay?

File: main.go (hunk 2 above)
Comment (ofalvai) [reply main.go]: Nice, I forgot those exist. Take a look now.

File: go.mod (hunk 3 above)
Comment (ofalvai) [reply go.mod:11]: I'm glad you asked! `go-steputils/v2` still has a few dependencies on `go-utils/v1`. I'm tracking them here: https://bitrise.atlassian.net/wiki/spaces/~932870153/pages/2452095043/Step+util+library+v2+migration+tracker

**Review by vshah23 — APPROVED** [review]

**Pull request merged**